planner/core/predicate_column_collector_test.go

```diff
@@ -0,0 +1,224 @@
+		},
+		{
+			// LogicalJoin
+			sql: "select * from t1 as x join t2 as y on x.a = y.a and x.b < 3 and y.c > 2",
+			res: []string{"t1.a", "t1.b", "t2.a", "t2.c"},
+		},
+		{
+			// LogicalJoin
+			sql: "select x.b, y.c, sum(x.c), sum(y.b) from t1 as x join t2 as y on x.a = y.a group by x.b, y.c order by x.b",
+			res: []string{"t1.a", "t1.b", "t2.a", "t2.c"},
+		},
+		{
+			// LogicalApply
+			sql: "select * from t1 where t1.b > all(select b from t2 where t2.c > 2)",
+			res: []string{"t1.b", "t2.b", "t2.c"},
+		},
+		{
+			// LogicalApply
+			sql: "select * from t1 where t1.b > (select count(b) from t2 where t2.c > t1.a)",
+			res: []string{"t1.a", "t1.b", "t2.b", "t2.c"},
+		},
+		{
+			// LogicalApply
+			sql: "select * from t1 where t1.b > (select count(*) from t2 where t2.c > t1.a)",
+			res: []string{"t1.a", "t1.b", "t2.c"},
+		},
+		{
+			// LogicalSort
+			sql: "select * from t1 order by c",
+			res: []string{"t1.c"},
+		},
+		{
+			// LogicalTopN
+			sql: "select * from t1 order by a + b limit 10",
+			res: []string{"t1.a", "t1.b"},
+		},
+		{
+			// LogicalUnionAll
+			sql: "select * from ((select a, b from t1) union all (select a, c from t2)) as tmp where tmp.b > 2",
+			res: []string{"t1.b", "t2.c"},
+		},
+		{
+			// LogicalPartitionUnionAll
+			sql: "select * from t3 where a < 15 and b > 1",
+			res: []string{"t3.a", "t3.b"},
+		},
+		{
+			// LogicalCTE
+			sql: "with cte(x, y) as (select a + 1, b from t1 where b > 1) select * from cte where x > 3",
+			res: []string{"t1.a", "t1.b"},
+		},
+		{
+			// LogicalCTE, LogicalCTETable
+			sql: "with recursive cte(x, y) as (select c, 1 from t1 union all select x + 1, y from cte where x < 5) select * from cte",
+			res: []string{"t1.c"},
+		},
+		{
+			// LogicalCTE, LogicalCTETable
+			sql: "with recursive cte(x, y) as (select 1, c from t1 union all select x + 1, y from cte where x < 5) select * from cte where y > 1",
+			res: []string{"t1.c"},
+		},
+		{
+			// LogicalCTE, LogicalCTETable
+			sql: "with recursive cte(x, y) as (select a, b from t1 union select x + 1, y from cte where x < 5) select * from cte",
+			res: []string{"t1.a", "t1.b"},
+		},
+	}
+
+	ctx := context.Background()
+	sctx := tk.Session()
+	is := dom.InfoSchema()
+	getColName := func(tblColID model.TableColumnID) (string, bool) {
+		tbl, ok := is.TableByID(tblColID.TableID)
+		if !ok {
+			return "", false
+		}
+		tblInfo := tbl.Meta()
+		for _, col := range tblInfo.Columns {
+			if tblColID.ColumnID == col.ID {
+				return tblInfo.Name.L + "." + col.Name.L, true
+			}
+		}
+		return "", false
+	}
+	checkPredicateColumns := func(lp plannercore.LogicalPlan, expected []string, comment string) {
+		tblColIDs := plannercore.CollectPredicateColumnsForTest(lp)
+		cols := make([]string, 0, len(tblColIDs))
+		for _, tblColID := range tblColIDs {
+			col, ok := getColName(tblColID)
+			require.Equal(t, true, ok, comment)
+			cols = append(cols, col)
+		}
+		sort.Strings(cols)
+		require.Equal(t, expected, cols, comment)
```

Comment:
`require.ElementsMatch()`

Reply:
done.

Reply:
If we use `ElementsMatch`, we no longer need to sort the slice. 😆

Reply:
That's cool. Updated.